src/sntp.rs

```diff
@@ -174,6 +195,17 @@ impl EspSntp {
             (*tv).tv_usec,
         );
     }
+
+    unsafe extern "C" fn sync_cb_for_subscribe(tv: *mut timeval) {
+        let duration: core::time::Duration = core::time::Duration::from_secs((*tv).tv_sec as u64);
+        let duration: core::time::Duration =
+            duration + core::time::Duration::from_micros((*tv).tv_usec as u64);
+
+        // FIXME: catch_unwind, so you don't panic across `extern "C"`
```

Comment:
What exactly is this FIXME `catch_unwind` supposed to mean? ESP IDF has zero support for unwinding the call stack anyway?